pkg/ignore/ignore.go

```diff
@@ -0,0 +1,127 @@
+package ignore
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+type DockerIgnorer struct {}
+
+func (b *DockerIgnorer) Ignore(config *api.Config) error {
+	// so, to duplicate the .dockerignore capabilities (https://docs.docker.com/reference/builder/#dockerignore-file)
+	// we have a flow that follows:
+	/*
+
+		0) First note, .dockerignore rules are NOT recursive (unlike .gitignore) .. you have to list subdir explicitly
+		1) Read in the exclusion patterns
+		2) Skip over comments (noted by #)
+		3) note overrides (via exclamation sign i.e. !) and reinstate files (don't remove) as needed
+		4) leverage Glob matching to build list, as .dockerignore is documented as following filepath.Match / filepath.Glob
+		5) del files
+
+                 1 to 4 is in getListOfFilesToIgnore
+
+	*/
+
+	filesToDel, lerr := getListOfFilesToIgnore(config)
+	if lerr != nil {
+		return lerr
+	}
+
+	if filesToDel == nil {
+		return nil
+	}
+
+	// delete compiled list of files 
+	for _, fileToDel := range filesToDel {
+		glog.V(5).Infof("attempting to remove file %s \n", fileToDel)
+		rerr := os.RemoveAll(fileToDel)
+		if rerr != nil {
+			glog.Errorf("error removing file %s because of %v \n", fileToDel, rerr)
+			return rerr
+		}
+	}
+
+	return nil
+}
+
+func getListOfFilesToIgnore(config *api.Config) (map[string]string, error) {
```

Comment:
@bparees and I discussed this previously in this PR.  I advocated for the `config` structure so the method signature looks more consistent with Build and Prepare, and allows for use of additional fields from `config` in the future without having to change the calls into Ignore.  As of the last discussion, we kept the parameter as is.